packages/js/src/ui/context/LocalizationContext.tsx

```diff
@@ -0,0 +1,55 @@
+import { ParentProps, createContext, createMemo, useContext } from 'solid-js';
+import { getNestedProperty } from '../helpers';
+
+export type Localization = {
+  inbox?: {
+    title?: string;
+  };
+  notification?: {
+    mainButtonText?: string;
+  };
+};
+
+type Path<T, K extends string = ''> = T extends string
+  ? K
+  : {
+      [P in keyof T]-?: Path<T[P], `${K}${K extends '' ? '' : '.'}${P & string}`>;
+    }[keyof T];
+
+type LocalizationPath = Path<Localization>;
+
+type LocalizationContextType = {
+  localized: (key: LocalizationPath) => string;
+};
+
+const LocalizationContext = createContext<LocalizationContextType | undefined>(undefined);
+
+type LocalizationProviderProps = ParentProps & { localization?: Localization };
+
+export const LocalizationProvider = (props: LocalizationProviderProps) => {
+  const localized = createMemo(() => (key: string) => {
+    const value = getNestedProperty(props.localization, key);
+    console.log(key, value);
+
+    return (value || '') as string;
+  });
+
+  return (
+    <LocalizationContext.Provider
+      value={{
+        localized: localized(),
+      }}
+    >
+      {props.children}
+    </LocalizationContext.Provider>
+  );
+};
+
+export function useLocalization() {
```

Comment:
We will need more of these so [feel free](https://github.com/clerk/javascript/blob/18807a454355d506ba034d4040ad0ece3d7a714b/packages/shared/src/react/hooks/createContextAndHook.ts#L20) :)

Reply:
Rule is on 3 :D